marconi-sidechain/doc/adr/01-sdd-computation.adoc

```diff
@@ -0,0 +1,203 @@
+
+The drawbacks:
+
+* requires a change to `cardano-node` in order to store those LedgerEvents. This implies lots of discussions with the `cardano-node` team to accept such a change.
+* needs the SPO to resync the relay node from scratch in order to save the `LedgerEvent` on disk. However, such a resync will only be required once, even after an upgrade on `marconi-sidechain`.
+* need to maintain a separate fork of the `cardano-node` in the event of the node team not agreeing to merge those changes
+
+=== 3. Index SDD from the state query client of the node-to-client protocol
+
+This alternative solution takes advantage of the only way currently to get the SDD from a `cardano-node` by using the state query client of the node-to-client protocol.
+However, the query will only return the latest SDD given the slot from which the Cardano node is synced to.
+In order to take advantage of this functionality, we would need to change the deployment method of `marconi-sidechain` to something as follows:
+
+. The SPO needs to stop the relay node and delete the `cardano-node` database
+. Start `marconi-sidechain` which will wait for the node to start
+. Start the relay node
+. Once Marconi has noticed that the node has started syncing, it will start querying the local query state for the SDD for each new epoch and index it in the database.
+
+The container diagram is the same as the link:#compute-sdd-by-tracking-the-ledgerstate[Compute SDD by tracking the LedgerState] solution.
+
+The benefits:
+
+* no change to `cardano-node` are required
+* uses the existing infrastructure to get the required information
+
+The drawbacks:
+
+* needs the SPO to resync the relay node from scratch
+* any indexing logic or database schema change to Marconi would require reindexing the local node (thus deleting the node database)
+* need to continuously poll the relay node when a new epoch occurs
+
+=== 4. Transform `marconi-sidechain` to a Cardano relay node
+
+`marconi-sidechain` could be changed to become _a_ Cardano node and replace the SPO's relay node.
+Then, it would have the same functionality as a relay (using node-to-node protocol), but also index
+the necessary for the Sidechain team.
+
+A container diagram of this solution would look like:
+
+image::http://www.plantuml.com/plantuml/png/jO_1JW8n48RlVOgw9qY4JOmdJyZgnSYoWNWcfPr0GzjfsccG6D_TLW8BYIPUl4sdt_pppIII29IiOJVQIPCKy2sHZmzr7SH-lD6qJUiK8KXq18x64ctjZN1bPXH96Rskc_jHLrflizVQYaWGSclLe2EX7OIn_vVy9_063xOCVIzWBIeOH8-tl5fsSwxE_zMGVp1szXcilSlCrA3SyIwWIbAx3I_ZrV2Iiw9ewSpEZXLVW1674pdL-Tb3nrI9rYQdfI28V9_FgtAplMkv3qWeqLolY5g1tEQwI9zg_kXznahwnNzl8uKniUI-BEVjQD4rHvaVyT-LcYkbJyQMXqXWLy_lD81GX1eHb7387T0Um-wy1KAV_yUJ9-1KikOB[]
+
+The benefits:
+
+* no change to `cardano-node` are required
+* only one computation of `LedgerState` is required
+
+The drawbacks:
+
+* uncertainty of capability to use `cardano-node` as a library
+* this version of `marconi-sidechain` would use the same amount of memory as the relay node.
```

Comment:
But we won't need the other relay-node, so we move from one relay-node + marconi-sidechain to "just" Marconi-relay-node. Which probably even better (resource-wise) than relay-node + marconi-sidechain.

Reply:
True :) Not really a drawback. I'll remove it.